Podverse/PVMediaPlayer.swift

```diff
@@ -317,7 +318,29 @@ class PVMediaPlayer: NSObject {
         
         var currentPlaybackTime = NSNumber(value: self.audioPlayer.progress)
         
-        MPNowPlayingInfoCenter.default().nowPlayingInfo = [MPMediaItemPropertyArtist: podcastTitle, MPMediaItemPropertyTitle: episodeTitle, MPMediaItemPropertyPlaybackDuration: self.duration, MPNowPlayingInfoPropertyElapsedPlaybackTime: currentPlaybackTime, MPNowPlayingInfoPropertyPlaybackRate: rate]
+        let podcastImage = Podcast.retrievePodcastImage(podcastImageURLString: item.podcastImageUrl, feedURLString: item.podcastFeedUrl, managedObjectID: nil, completion: { _ in
+            
+            guard let item =  self.nowPlayingItem else {
+                return
+            }
+            
+            var currentPlaybackTime = NSNumber(value: self.audioPlayer.progress)
+            let rate = self.audioPlayer.rate
+        
+            if let podcastImageUrlString = item.podcastImageUrl, let podcastImageUrl = URL(string: podcastImageUrlString) {
+                if let data = try? Data(contentsOf: podcastImageUrl), let image = UIImage(data: data) {
```

Comment:
Something seems a bit off here. This looks like a synchronous call inside the completion block of retrievePodcastImage which should have already retrieved the image. I think there is something wrong with the retrievePodcastImage helper function. it either returns the image from disk or a completionBlock with a nil argument. Instead it should be calling the completionBlock with the already received image (either using sd_webImage or making an async call to fetch the data). And when all that is done, set the podcast image inside here. 
What is going on here is a completely different fetching of an image to set the MPNowPlayingInfoCenter info which is not incorrect, its just in the wrong spot.

Reply:
Hmmm, sorry I may need some help with cleaning this one up. 

My thinking was that since I don't know if .retrievePodcastImage will synchronously return the image or return it in the callback, I end up setting the MPNowPlayingInfoCenter both after .retrievePodcastImage and within it's callback, since I need the MPNowPlayingInfoCenter to load immediately in either case...